src/Shared/AssemblyNameExtension.cs

```diff
@@ -958,5 +994,25 @@ internal bool PartialNameCompare(AssemblyNameExtension that, PartialComparisonFl
             }
             return true;
         }
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            if (asAssemblyName != null)
```

Comment:
Probably nice to have it sync with [https://github.com/Microsoft/msbuild/blob/master/src/Shared/AssemblyUtilities.cs#L62](https://github.com/Microsoft/msbuild/blob/master/src/Shared/AssemblyUtilities.cs#L62)